src/aria/widgets/controllers/TextDataController.js

```diff
@@ -222,6 +256,104 @@ module.exports = Aria.classDefinition({
                 report : report,
                 arg : arg
             });
+        },
+
+        /**
+         * Retrieves the requested error message from different sets of configuration ordered by precedence.
+         *
+         * <p>
+         * An error message value is requested from an this error message name.
+         * </p>
+         *
+         * <p>
+         * There can be three levels of configuration for the error messages of a widget, enumerated here by order of precedence:
+         * <ol>
+         *  <li>local: in the configuration of the widget's instance</li>
+         *  <li>global: in the application's environment configuration</li>
+         *  <li>hardcoded: in internal resources; it is used as a fallback</li>
+         * </ol>
+         * </p>
+         *
+         * <p>
+         * As soon as a configuration contains a non-void value for the requested error message, this one is used for the return value.
+         * </p>
+         *
+         * <p>
+         * Note that since there are hardcoded values, there will always be an error message for a valid message name. However, if the message name is not supported, an <em>unefined</em> value is returned in the end.
+         * </p>
+         *
+         * @param {String} errorMessageName The name of the error message to retrieve (it is the key to the requested message in the collection).
+         *
+         * @return {String} The retrieved error message if any, <em>undefined</em> if the message name is not supported.
+         */
+        getErrorMessage : function (errorMessageName) {
```

Comment:
This function could be generalized and moved in a utils class, since the configuration and logical parts are well decoupled.